### PR TITLE
refactor CORS settings

### PR DESCRIPTION
**Pull request by bacongobbler**
this refactoring allows us to better test the default CORS settings used
by the brigade API.

### CODE REVIEW

File: brigade-api/cmd/brigade-api/main_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"net/http/httptest"
+	"testing"
+
+	"github.com/emicklei/go-restful"
+)
+
+func dummy(req *restful.Request, resp *restful.Response) { io.WriteString(resp.ResponseWriter, "dummy") }
+
+func TestDefaultCORSFilter(t *testing.T) {
+	buf := bytes.NewBuffer([]byte{})
```
Comment (bacongobbler) [inline brigade-api/cmd/brigade-api/main_test.go]: TODO: delete this

**Comment by bacongobbler** [comment]
NOTE: this doesn't actually change anything, just shifts things around.

File: brigade-api/cmd/brigade-api/main.go [hunk 2]
```diff
@@ -20,6 +20,14 @@ import (
 )
 
 var (
+	// DefaultCORS defines the default CORS settings for the API
+	DefaultCORS = restful.CrossOriginResourceSharing{
+		AllowedHeaders: []string{"Content-Type", "Accept"},
```
Comment (radu-matei) [inline brigade-api/cmd/brigade-api/main.go]: @bacongobbler - does this allow requests from all origins by default?

Comment (bacongobbler) [reply brigade-api/cmd/brigade-api/main.go]: Yep! I'm not sure if that's a good default, but that's the default setting from go-restful.

**Comment by bacongobbler** [comment]
will rebase then merge :) thanks everyone for the review!

**Comment by mumoshu** [comment]
@bacongobble Hey! Any chance you could merge this?

**Comment by radu-matei** [comment]
IIRC, this change was originally intended for Kashti - do we still need this, are there other scenarios where this would be useful?

**Comment by radu-matei** [comment]
Closing, but feel free to re-open if it turns out this is indeed needed again

Thanks!

**Pull request closed without merging**